### PR TITLE
feat: add customSvg attribute support #124

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #124 

## Summary:

The code diff introduces the following updates to `auro-icon.js` and `baseIcon.js`:

1. **New `customSvg` Attribute**:
   - Added a `customSvg` attribute of type `Boolean`.
   - Allows rendering a custom SVG inside the default slot when `true`.

1. **Default Initialization**:
   - Set `customSvg` to `false` in `privateDefaults()`.

1. **Template Update**:
   - Conditionally renders `svg` slot content based on `customSvg`.
   - Renders custom SVG if `customSvg` is `true`, otherwise defaults to `svg`.

1. **Conditional Icon Fetching**:
   - Added a check for `customSvg` before fetching the icon.
   - If `customSvg` is `false`, it fetches the icon based on `category` and `name`.

1. **SVG Assignment**:
   - If the icon fetch is successful, assigns the fetched SVG to `this.svg`.
   - If the icon fetch fails, parses and assigns an error SVG to `this.svg`.

These changes ensure that the icon fetch and assignment only occur when `customSvg` is `false`, and support custom SVG content via the new `customSvg` attribute and slot.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

## Verification

- [x] Existing auro-icon examples work correctly
- [x] New demo example on the API page demonstrating the new feature work correctly

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: apiExamples/customSvg.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
...
+</auro-icon>
+<auro-icon customSvg error>
+  <svg slot="svg" aria-labelledby="pin-trip-filled__desc" class="ico_squareLarge" role="img" viewBox="0 0 24 24" part="svg" style="min-width: var(--auro-size-lg, var(--ds-size-300, 1.5rem)); height: var(--auro-size-lg, var(--ds-size-300, 1.5rem)); fill: currentcolor;"><title></title><desc id="pin-trip-filled__desc">drop pin with circles.</desc><path d="M10.045 3.345a.75.75 0 0 1 .785-.714l.05.003a.75.75 0 0 1-.082 1.498l-.04-.002a.75.75 0 0 1-.713-.785m-1.217.22a.75.75 0 0 1-.357 1l-.034.016a.75.75 0 0 1-.655-1.35l.047-.022a.75.75 0 0 1 .999.357m3.949.186a.75.75 0 0 1 1.012-.318l.045.023a.75.75 0 0 1-.703 1.326l-.035-.019a.75.75 0 0 1-.319-1.012M6.508 5.057a.75.75 0 0 1 .2 1.041l-.01.017a.75.75 0 1 1-1.246-.836l.014-.022a.75.75 0 0 1 1.042-.2m8.577.22a.75.75 0 0 1 1.038.218l.028.044a.75.75 0 0 1-1.264.808l-.02-.032a.75.75 0 0 1 .218-1.038m6.02 7.014c0-2.789-2.44-4.88-4.88-4.88s-4.881 2.091-4.881 4.88q0 2.559 4.11 8.496l.199.285.055.068a.697.697 0 0 0 1.088-.068q4.31-6.16 4.309-8.781m-6.275 0a1.394 1.394 0 1 1 2.789 0 1.394 1.394 0 0 1-2.789 0M4.635 10.704a1.74 1.74 0 1 0 0-3.48 1.74 1.74 0 0 0 0 3.48"></path></svg>
+</auro-icon>
```
Comment (jordanjones243) [inline apiExamples/customSvg.html:9]: Add line at end of file

File: apiExamples/basic.html [hunk 2]
```diff
@@ -1 +1 @@
-<auro-icon category="interface" name="pin-trip"></auro-icon>
+<auro-icon category="interface" name="pin-trip"></auro-icon>
```
Comment (jordanjones243) [inline apiExamples/basic.html:1]: Add line end of file

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Need to run a build for doc changes

**Review by jordanjones243 — CHANGES_REQUESTED** [review]
Something is also wrong with the CTA buttons

**Review by blackfalcon — APPROVED** [review]

**Review by jordanjones243 — APPROVED** [review]

**Pull request merged**